mango-core/src/main/scala/org/bdgenomics/mango/models/TrackedLayout.scala

```diff
@@ -117,6 +141,33 @@ class OrderedTrackedLayout[T: ClassTag](values: Traversable[(ReferenceRegion, T)
     }
   }
 
+  private def findPairsAndAddToTrack(readPairs: Map[String, Seq[(ReferenceRegion, T)]]) {
+    readPairs.foreach {
+      p =>
+        {
+          val recs: Seq[(ReferenceRegion, T)] = p._2
+          val trackOption: Option[Track] = TrackTimers.FindConflict.time {
+            trackBuilder.find(track => !track.conflicts(recs))
+          }
+          val track: Track = trackOption.getOrElse(addTrack(new Track()))
+
+          recs.foreach {
+            rec: (ReferenceRegion, T) =>
+              {
+                if (!track.conflicts(rec)) {
+                  track += rec
+                  trackBuilder -= track
```

Comment:
Why do we need the:

```
trackBuilder -= track
trackBuilder += track
```

I feel like @erictu explained this to me a while ago.